src/overlays/actors/ovl_En_Pp/z_en_pp.c

```diff
@@ -64,116 +127,1500 @@ const ActorInit En_Pp_InitVars = {
+                            this->actor.world.rot.z, limbIndex + 7);
+                this->deadBodyPartsSpawnedCount++;
+                if (this->deadBodyPartsSpawnedCount >= 6) {
+                    this->action = EN_PP_ACTION_DONE_SPAWNING_BODY_PARTS;
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B1F940.s")
+void EnPp_Draw(Actor* thisx, PlayState* play) {
+    EnPp* this = THIS;
+    MtxF mtxF;
+    Vec3f pos;
+    f32 scale;
+    f32 alpha;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B1FAD0.s")
+    func_8012C2DC(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnPp_OverrideLimbDraw, EnPp_PostLimbDraw, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B1FC7C.s")
+    if (this->deadBodyPartDrawDmgEffCount != 0) {
+        scale = 0.4f;
+        if (EN_PP_GET_TYPE(&this->actor) == EN_PP_TYPE_BODY_PART_BODY) {
+            scale = 0.6f;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B1FF20.s")
+        Actor_DrawDamageEffects(play, &this->actor, this->deadBodyPartDrawDmgEffPos, this->deadBodyPartDrawDmgEffCount,
+                                scale, scale, 1.0f, ACTOR_DRAW_DMGEFF_BLUE_FIRE);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B20030.s")
+    if (this->drawDmgEffTimer != 0) {
+        alpha = this->drawDmgEffTimer * 0.05f;
+        if ((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
+            (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) {
+            this->drawDmgEffScale += 0.3f;
+            if (this->drawDmgEffScale > 0.5f) {
+                this->drawDmgEffScale = 0.5f;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B202B8.s")
+            Math_ApproachF(&this->drawDmgEffFrozenSteamScale, this->drawDmgEffScale, 0.1f, 0.04f);
+        } else {
+            this->drawDmgEffScale = 0.8f;
+            this->drawDmgEffFrozenSteamScale = 0.8f;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B203BC.s")
+        Actor_DrawDamageEffects(play, &this->actor, this->bodyPartsPos, ARRAY_COUNT(this->bodyPartsPos),
+                                this->drawDmgEffScale, this->drawDmgEffFrozenSteamScale, alpha, this->drawDmgEffType);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/func_80B20668.s")
+    if (this->floorPolyForCircleShadow != NULL) {
+        if ((this->action != EN_PP_ACTION_MASK_DETACH) && (this->action < EN_PP_ACTION_DEAD)) {
+            OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pp/EnPp_Update.s")
+            func_8012C448(play->state.gfxCtx);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0x00, 0x00, 0x00, 0xFF);
```

Comment:
```suggestion
            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0 255);
```
I believe these can be decimal